src/common.c

```diff
@@ -363,15 +363,14 @@ void compute_powers(fr_t *out, const fr_t *x, uint64_t n) {
 /**
  * Deserialize a blob (array of bytes) into a polynomial (array of field elements).
  *
- * @param[out] p    The output polynomial (array of field elements)
- * @param[in]  blob The blob (an array of bytes)
+ * @param[out]  p           The output polynomial (array of field elements)
+ * @param[in]   blob        The blob (an array of bytes)
+ * @param[in]   num_fields  The number of field elements in the blob
  */
-C_KZG_RET blob_to_polynomial(Polynomial *p, const Blob *blob) {
+C_KZG_RET blob_to_polynomial(fr_t *p, const uint8_t *blob, size_t num_fields) {
```

Comment:
Hm. Why was this function made generic? Seems like we always use it with `n=FIELD_ELEMENTS_PER_BLOB` anyway.

BTW, IMO `num_fields` as a var name can be improved. "fields" is not a common way to call "field elements". I think `num_fields -> n` would make things better, but perhap we can just remove this change altogether if we don't need the function to be generic.

Reply:
This function is defined in common, but references Blob and Polynomial. Blob should be defined in eip4844.h and Polynomial shouldn't be exposed. This is the best way I thought to handle this.

Reply:
Hmm, I see.
I think Blob and FIELD_ELEMENTS_PER_BLOB should be moved to common, since they are used both by 4844 and 7594. I did this in e4d6c1cc.

Then in 551b7c53, I restored `blob_to_polynomial()` closer to what it once was. To minimize the diffs of audited 4844 code.